webtau-http/src/main/java/com/twosigma/webtau/http/Http.java

```diff
@@ -535,6 +535,10 @@ private HttpResponse request(String method, String fullUrl,
             connection.setRequestMethod(method);
             connection.setRequestProperty("Content-Type", requestBody.type());
             connection.setRequestProperty("Accept", requestBody.type());
+
+            if (requestHeader == null) {
+                throw new IllegalArgumentException("Request header is null, check your header provider is not returning null");
+            }
```

Comment:
could you please put an empty line here to separate logical blocks?
also wonder if we could spare the whole url connection creation.

Reply:
Done.

Reply:
what about "moving it to the top" part to avoid creating connection in case of bad header?